### PR TITLE
feat: sharable organization

### PR DESCRIPTION
**Pull request by oneofthezombies**
# PR Template

## Checklist

- [ ] E2E finished.
- [ ] Local, Self Hosted and Development environment test finished.
- [v] Local, E2E, Self Hosted, Development and Production env variable checked.
- [v] (Backend only) DB migration file creation checked.
- [v] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

### CODE REVIEW

File: packages/typescript-private/console-host-agent/src/http-specs/private-device.ts [hunk 1]
```diff
@@ -35,6 +35,9 @@ export class CreateDeviceRequestBody implements Pick<Required<Device>, 'serial'
   @IsNumber()
   @IsNotEmpty()
   isVirtual!: number;
+
+  @IsString()
+  memory!: string;
```
Comment (Ikalli) [inline packages/typescript-private/console-host-agent/src/http-specs/private-device.ts:40]: Does it need nullable or `IsNotEmpty()` here?

File: packages/typescript-private/console-host-agent/src/http-specs/private-device.ts [hunk 2]
```diff
@@ -110,6 +116,9 @@ export class UpdateDeviceRequestBody
   @Type(() => Number)
   resolutionHeight!: number;
 
+  @IsString()
+  memory!: string;
```
Comment (Ikalli) [inline packages/typescript-private/console-host-agent/src/http-specs/private-device.ts:120]: Does it need nullable or `IsNotEmpty()` here?

File: projects/console-web-server/src/db/migrations/local/1696589377345-typeorm-migration.ts
Comment (Ikalli) [inline projects/console-web-server/src/db/migrations/local/1696589377345-typeorm-migration.ts:1]: Migration files are missing:
- e2e
- development
- production
- self-hosted

**Review by Ikalli — CHANGES_REQUESTED** [review]

File: packages/typescript-private/console-host-agent/src/http-specs/private-device.ts (hunk 1 above)
Comment (oneofthezombies) [reply packages/typescript-private/console-host-agent/src/http-specs/private-device.ts:40]: @Ikalli You're right. But I didn't implement the memory of the Macos device. 
Beaming Face with Smiling Eyes

File: packages/typescript-private/console-host-agent/src/http-specs/private-device.ts (hunk 2 above)
Comment (oneofthezombies) [reply packages/typescript-private/console-host-agent/src/http-specs/private-device.ts:120]: @Ikalli And it hasn't been verified that there's a value for the emulator or every case. In this case, I didn't want the device creation to fail on the console. Face Screaming in Fear

File: projects/console-web-server/src/db/migrations/local/1696589377345-typeorm-migration.ts
Comment (oneofthezombies) [reply projects/console-web-server/src/db/migrations/local/1696589377345-typeorm-migration.ts:1]: @Ikalli Thank you. i added. Raising Hands

**Pull request merged**